src/nxt_fs.c

```diff
@@ -62,7 +63,7 @@ nxt_fs_mkdir_parent(const u_char *path, mode_t mode)
     ret = NXT_OK;
 
     ptr = strrchr(dir, '/');
-    if (nxt_slow_path(ptr == NULL)) {
+    if (nxt_slow_path(ptr == NULL || ptr == dir)) {
```

Comment:
You mean if someone tried to create the socket or whatever in the root directory?

Reply:
Yep.  That's unlikely.  However, it's necessary for using recursion, since at some point it'll arrive at the parent-most dir, which is of course `/usr`, and when trying to create its dirname, it'll try `/`.